Celeste.Mod.mm/Patches/Monocle/Tracker.cs

```diff
@@ -123,5 +123,109 @@ private static List<Type> GetSubclasses(Type type) {
+                        if (!TrackedEntityTypes.TryGetValue(subtype, out List<Type> value)) {
+                            value = new List<Type>();
+                            TrackedEntityTypes.Add(subtype, value);
+                        }
+                        value.AddRange(TrackedEntityTypes.TryGetValue(trackedAs != null && trackedAs.IsAssignableFrom(type)
+                            ? trackedAs : type, out List<Type> list) ? list : new List<Type>());
+                        TrackedEntityTypes[subtype] = value.Distinct().ToList();
+                    }
+                }
+            }
+            else if (typeof(Component).IsAssignableFrom(type)) {
+                StoredComponentTypes.Add(type);
+                if (!type.IsAbstract) {
+                    if (!TrackedComponentTypes.TryGetValue(type, out List<Type> value)) {
+                        value = new List<Type>();
+                        TrackedComponentTypes.Add(type, value);
+                    }
+                    value.AddRange(TrackedComponentTypes.TryGetValue(trackedAs != null && trackedAs.IsAssignableFrom(type)
+                        ? trackedAs : type, out List<Type> list) ? list : new List<Type>());
+                    TrackedComponentTypes[type] = value.Distinct().ToList();
+                }
+                foreach(Type subtype in subtypes) {
+                    if (!subtype.IsAbstract) {
+                        if (!TrackedComponentTypes.TryGetValue(subtype, out List<Type> value)) {
+                            value = new List<Type>();
+                            TrackedComponentTypes.Add(subtype, value);
+                        }
+                        value.AddRange(TrackedComponentTypes.TryGetValue(trackedAs != null && trackedAs.IsAssignableFrom(type)
+                            ? trackedAs : type, out List<Type> list) ? list : new List<Type>());
+                        TrackedComponentTypes[subtype] = value.Distinct().ToList();
+                    }
+                }
+            }
+            else {
+                throw new Exception("Type '" + type.Name + "' cannot be TrackedAs because it does not derive from Entity or Component");
+            }
+            RefreshTracker(type);
+        }
+
+        public static void RefreshTracker(Type type) {
+            if (typeof(Entity).IsAssignableFrom(type) && !Engine.Scene.Tracker.Entities.ContainsKey(type)) {
+                Engine.Scene.Tracker.Entities.Add(type, new List<Entity>());
+            }
+            else if (typeof(Component).IsAssignableFrom(type) && !Engine.Scene.Tracker.Components.ContainsKey(type)) {
+                Engine.Scene.Tracker.Components.Add(type, new List<Component>());
+            } else {
+                throw new Exception("Type '" + type.Name + "' cannot be TrackedAs because it does not derive from Entity or Component");
+            }
+            RefreshTrackerLists();
+        }
+
+        public static void RefreshTracker() {
```

Comment:
Is there any point to this being public? Only case where you'd need to refresh the tracker is when calling `AddTypeToTracker` anyway 🤔

Reply:
Allows mods to manually refresh the tracker of another scene, since it wouldnt happen automatically.
However it might make sense to automatically refresh the tracker on scene change to avoid having to manually call it?

Reply:
Well in that case this method should definitely be an instance method (called `Refresh()`), otherwise I don't see a sane way to use it on a different scene. This only seems useful in *very* niche cases, (creating a Level inside a Level??) but ig it doesn't hurt (With a comment with `///`)

Reply:
Oh yeah. It definitly shouldnt be static

Reply:
My main question would be how do I know if there's a scene change at any time? Besides storing the "old" one in a variable and check against it every frame?